bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/handler/RoombaHandler.java

```diff
@@ -0,0 +1,575 @@
+                    RawMQTT.PasswdPacket passwdPacket = new RawMQTT.PasswdPacket(response);
+                    String password = passwdPacket.getPassword();
+
+                    if (password != null) {
+                        config.password = password;
+
+                        Configuration configuration = editConfiguration();
+
+                        configuration.put("password", password);
+                        updateConfiguration(configuration);
+
+                        logger.debug("Password successfully retrieved");
+                    }
+                }
+            }
+
+            if (config.password.isEmpty()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                        "Authentication on the robot is required");
+                scheduleReconnect();
+                return;
+            }
+
+            // BLID is used as both client ID and username. The name of BLID also came from Roomba980-python
+            MqttBrokerConnection connection = new MqttBrokerConnection(config.ipaddress, RawMQTT.ROOMBA_MQTT_PORT, true,
+                    blid);
+
+            this.connection = connection;
+
+            // Disable sending UNSUBSCRIBE request before disconnecting becuase Roomba doesn't like it.
+            // It just swallows the request and never sends any response, so stop() method never completes.
+            connection.setUnsubscribeOnStop(false);
+            connection.setCredentials(blid, config.password);
+            connection.setTrustManagers(RawMQTT.getTrustManagers());
+            // 1 here actually corresponds to MQTT qos 0 (AT_MOST_ONCE). Only this value is accepted
+            // by Roomba, others just cause it to reject the command and drop the connection.
+            connection.setQos(1);
+            // MQTT connection reconnects itself, so we don't have to call scheduleReconnect()
+            // when it breaks. Just set the period in ms.
+            connection.setReconnectStrategy(
+                    new PeriodicReconnectStrategy(RECONNECT_DELAY_SEC * 1000, RECONNECT_DELAY_SEC * 1000));
+            connection.start().exceptionally(e -> {
+                connectionStateChanged(MqttConnectionState.DISCONNECTED, e);
+                return false;
+            }).thenAccept(v -> {
+                if (!v) {
+                    connectionStateChanged(MqttConnectionState.DISCONNECTED, new TimeoutException("Timeout"));
+                } else {
+                    connectionStateChanged(MqttConnectionState.CONNECTED, null);
+                }
+            });
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            scheduleReconnect();
+        }
+    }
+
+    private synchronized void disconnect() {
+        if (reconnectReq != null) {
+            reconnectReq.cancel(false);
```

Comment:
The compiler warning can be avoided by storing the reference to a local variable and do the null check on that. Same for below.

Reply:
ACK